mindsdb/integrations/handlers/llama_index_handler/llama_index_handler.py

```diff
@@ -0,0 +1,165 @@
+from mindsdb.integrations.libs.base import BaseMLEngine
+
+from mindsdb.utilities.log import get_log
+
+
+logger = get_log("integrations.llama_index_handler")
+
+
+
+class LlamaIndexHandler(BaseMLEngine):
+    """
+    Integration with the LlamaIndex Python Library
+    """
+    name = 'llama_index'
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+        self.default_index_class  = 'GPTVectorStoreIndex'
+        self.supported_index_class = ['GPTVectorStoreIndex']
+        self.supported_query_engine = ['as_query_engine'] 
+        self.default_reader = 'DFReader'
+        self.supported_reader = ['DFReader','SimpleWebPageReader']
+  
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        
+        
+        if 'using' not in args:
+            raise Exception("LlamaIndex engine requires a USING clause! Refer to its documentation for more details.")
+
+        if args['using']['reader'] == 'SimpleWebPageReader':
+            if 'source_url_link' not in args['using']:
+                raise Exception("LlamaIndex engine requires a source_url_link parameter.Refer to its documentation for more details.")
+
+        if 'openai_api_key' not in args['using']:
+            raise Exception("LlamaIndex engine requires a openai_api_key parameter.Refer to its documentation for more details.")
+
+        if 'index_class' not in args['using']:
+            args['using']['index_class'] = self.default_index_class
+        elif args['using']['index_class'] not in self.supported_index_class:
+            raise Exception(f"Invalid index class argument. Please use one of {self.supported_index_class}")
+        
+        if 'query_engine' not in args['using']:
+            args['using']['query_engine'] = self.default_query_engine
+        elif args['using']['query_engine'] not in self.supported_query_engine:
+            raise Exception(f"Invalid operation mode. Please use one of {self.supported_query_engine}")
+
+        if 'reader' not in args['using']:
+            args['using']['reader'] = self.default_reader
+        elif args['using']['reader'] not in self.supported_reader:
+            raise Exception(f"Invalid operation mode. Please use one of {self.supported_query_engine}")
+
+       
+        documents_df_reader = []
+        documents_url_reader = []
+        pred = None
+        if args['using']['reader'] == 'DFReader':
+            for row in df.itertuples():
+                doc_str = ", ".join([str(entry) for entry in row])
+                documents_df_reader.append(Document(doc_str))  
+            pred = documents_df_reader   
+      
+        elif args['using']['reader'] == 'SimpleWebPageReader':
+            SimpleWebPageReader = download_loader("SimpleWebPageReader")
+            documents_url_reader = SimpleWebPageReader(html_to_text=True).load_data([args['using']['source_url_link']])
+            pred = documents_url_reader 
+    
```

Comment:
Seems like an `else: raise Exception()` is needed here. Otherwise we store `None` and it would break at prediction time.

Reply:
Thanks for suggestion @paxcema, I have made the suggested changes